oops/oops.go

```diff
@@ -130,6 +130,34 @@ func Frames(err error) [][]Frame {
 	return parsedStacks
 }
 
+// DropStackFrames drops numFrames from the stack trace and returns a new copy of the error
+func DropStackFrames(err error, numFrames uint) error {
+	var e *oopsError
+	if ok := As(err, &e); !ok {
+		return err
+	}
+	st := e.stack
+	if st == nil {
+		return err
+	}
+	numLeftoverFrames := len(st.frames) - int(numFrames)
+	if numLeftoverFrames < 0 {
+		return err
+	}
```

Comment:
to confirm the logic: we do no-op here if you specified a skip too deep because it's probably user error?

Reply:
yeah

Reply:
could you add a comment describing this?